### ISSUE TITLE
Woocommerce Product Reviews Widget causing issue with Divi Theme Builder

### ISSUE THREAD

**Issue by courtney666** [woothemes/woocommerce#31022]
When using Woocommerce Product Reviews Widget in Footer Widget and then using the Divi > Theme Builder to build something the builder keeps on loading and stays stuck. The issue has been tested with multiple themes and also on fresh install, it has been same everywhere.

**Comment by rodelgc** [comment]
Hi @courtney666,

Thank you for submitting the issue. However, you didn’t fill out the details of the bug report template that we ask for. Without these details, we can’t fully evaluate this issue. Please provide us with the information requested so we could take a look further.

**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

**Comment by courtney666** [comment]
Hello,

Here is how u can test it.

Have latest divi plugin with you.

Then go to theme builder: https://prnt.sc/1x57i5c

There click on add a new template and then Create Template

After that add custom body > build custom body: https://prnt.sc/1x57j8x

Once done then the page keeps on loading like this: https://prnt.sc/1x57kr4

Upon checking logs, these were the logs: https://prnt.sc/1xaocp2

Recent Product Reviews Widget is causing that issue as per the logs, so on removing this widget: https://prnt.sc/1xaof2x

All things get solved.

NOTE: THE BUG HAPPENS IN EVEN DEFAULT WORDPRESS THEME, TRIED EVEN ON FRESH INSTALL AND THE BUG IS THERE

(I have a screen recording for the same too, but where can I share it? I don't want to share it in public.)

**Comment by rodelgc** [comment]
Thank you very much for the feedback, @courtney666!

It appears that the issue you're experiencing is caused by Divi, since most of the steps you mentioned were using their theme builder. It also appears that the Recent Product Reviews Widget is from them as well, and no longer part of WooCommerce Core.

Please contact them directly for further help.

For the meantime, I'll close this issue but feel free to leave a comment if there's something I missed so we could take another look.

**Comment by courtney666** [comment]
Hello,

Are you guys serious?

Recent Product Reviews Widget is from Woocommerce, lol

I shared you logs, it shows the woocommerce code, see this: https://prnt.sc/1xaocp2

**Comment by rodelgc** [comment]
Hi @courtney666, apologies for the confusion. You're right, Recent Product Reviews is one of the Legacy Widgets. I was paying more attention to the Block ones. 

I'm reopening this issue with the `needs developer feedback` label so that our Core devs could take a closer look upon looking at the logs you shared.

**Comment by courtney666** [comment]
Thanks a lot, I hope that it gets resolved soon, for now since removing that widget is not causing any issues, we are not using that widget.

**Comment by courtney666** [comment]
When do topics usually get checked by developers?

**Comment by lanej0** [comment]
Hi @courtney666,

Just wanted to follow up with you that we haven't forgotten about this and that it's in the queue for our developers to look at.  We're working through a little bit of a backlog at the moment, but hopefully should have some updates shortly.

**Comment by courtney666** [comment]
Hello,

Hope to see it fixed, it’s been a long time already.

Regards,
Courtney Brooks

**Comment by barryhughes** [comment]
Hi @courtney666,

Sorry for the delay. What I've tried so far:

- I installed Divi 4.14.8.
- I enabled classic widgets and added the Recent Product Reviews widget to the footer.
- I went to **Divi ▸ Theme Builder ▸ Add Global Body ▸ Build Global Body**.
    - The above seems just a little different to what you described...
    - ...I'm ascribing that to recent changes in Divi, but let me know if I'm mistaken.
- It did take quite a while for the editor to load, but it did eventually load.
- I successfully added a text module and a few other things.

The above worked even if `WP_DEBUG_DISPLAY` was set to true. Is it possible this is solved as of a recent Divi release (what version are you using) and have you found any method of reproducing this *without* Divi also being present?

---

Besides being unable to replicate, I wanted to comment on the captured error (mostly for our own benefit, so it's faster to pick this up again).

It seems pretty clear that, when this problem happens, the `$product` var [used here](https://github.com/woocommerce/woocommerce/blob/6.3.0/plugins/woocommerce/templates/content-widget-reviews.php#L29) is set to `false`, leading to the _call to a member function on bool_ error ... yet, upstream of this code, we explicitly fetch [comments associated with products](https://github.com/woocommerce/woocommerce/blob/6.3.0/plugins/woocommerce/includes/widgets/class-wc-widget-recent-reviews.php#L60-L68). That makes me wonder if things are breaking because another component is interfering with queries.

We can definitely add a safeguard, but if the same problem happens outside of the builder environment it would still lead to a broken or incomplete-looking widget.
